lib/delayed/backend/base.rb

```diff
@@ -139,6 +139,10 @@ def max_run_time
         end
       end
 
+      def destroy_failed_jobs
+        payload_object.destroy_failed_jobs if payload_object.respond_to?(:destroy_failed_jobs)
+      end
+
```

Comment:
I am also thinking we make the payload object implement `destroy_failed_jobs?` instead of `destroy_failed_jobs`.